app/controllers/api/v2/accredited_body_training_provider_courses_controller.rb

```diff
@@ -0,0 +1,37 @@
+module API
+  module V2
+    class AccreditedBodyTrainingProviderCoursesController < ApplicationController
+      before_action :build_recruitment_cycle, :build_accredited_body, :build_training_provider
+
+      def index
+        authorize accredited_body, :can_list_training_providers?
+
+        course_scope = accredited_body.current_training_provider_courses(training_provider.id)
+        courses = CoursePolicy::TrainingProviderCourses.new(current_user, course_scope).resolve
+        render jsonapi: courses, include: params[:include]
+      end
+
+    private
+
+      attr_reader :recruitment_cycle, :accredited_body, :training_provider
+
+      def build_recruitment_cycle
+        @recruitment_cycle = RecruitmentCycle.find_by(
+          year: params[:recruitment_cycle_year],
+        ) || RecruitmentCycle.current_recruitment_cycle
+      end
+
+      def build_accredited_body
```

Comment:
```suggestion
      def build_provider
```

Reply:
I think the naming is important in here as the controller is dealing with two different providers so `accredited_body` & `training_provider` makes the distinction. Although they are both instances of `Provider` the distinction is important in this context.

Reply:
https://github.com/DFE-Digital/teacher-training-api/blob/624e1b77dcca3ef4d8794c687e47aece3347fa19/app/controllers/api/v2/accredited_provider_training_providers_controller.rb#L22-L26

something feels off with naming things

Accredited`Body`TrainingProviderCoursesController
Accredited`Provider`TrainingProvidersController

can we settle on `accredited_provider`

Reply:
I did settle on that in my last PR but since then the decision was made to settle on accredited body. I've got a ticket in my backlog to make it consistent throughout which I'll be pushing for the next sprint. I can change the `...Provider` one in this PR but there are references accredited|accrediting provider all over the place that still need fixing.